monster.py

```diff
@@ -2,31 +2,22 @@
 
 from moving_thing import MovingThing
 from abc import ABCMeta, abstractmethod
+from collections import namedtuple
 
+Point = namedtuple('Point', ['y', 'x'])
 
 class Monster(MovingThing, metaclass=ABCMeta):
     def follow_player(self):
-        nx, ny = self.x, self.y
-        if self.y == self.room.player.y and self.room.player.x == self.x:
-            return
-
-        if self.room.player.x > self.x:
-            nx = self.x + 1
-        elif self.room.player.x < self.x:
-            nx = self.x - 1
-
-        if self.room.player.y > self.y:
-            ny = self.y + 1
-        elif self.room.player.y < self.y:
-            ny = self.y - 1
-
-        if self.x != nx and self.y != ny:
-            if random.randrange(2):
-                nx = self.x
-            else:
-                ny = self.y
-
-        self.start_moving(nx, ny)
+        neighbours = [n for n in {Point(self.y - 1, self.x),
```

Comment:
tuto set urcite nie je rychlejsi, nie je to boost, ... pouzi tuple

(je to ako keby si pouzil list, ale kedze zoznam nemusime menit, mozeme si dovolit iterovat cez immutable strukturu - a tuple je v pythone este trosku rychlejsi ako list)